### PR TITLE
Parsing and cleaning atoc data

### LINKED ISSUE #293: Parse atoc data (proof of concept)
Develop code that allows us to parse the master station names (`.msn`) and small subset of the timetable data (`.mca`) into pandas dataframes as a proof of concept.

Code currently in `atoc` branch

### LINKED ISSUE #299: Parse all atoc data
Following from setting up the POC to parse and extract relevant data for a small subset of atoc data (#293) - test and edit the code to handle the full file.

### PR DESCRIPTION
**Pull request by nkshaw23**
Refactored code to parse and clean the atoc data.

Closes #299 
Closes #293

### CODE REVIEW

**Comment by james-westwood** [comment]
@nkshaw23 I am pulling the code now to check.

**Comment by james-westwood** [comment]
@nkshaw23 

These are the requirements I am checking against (from the tickets)
- parse the master station names (.msn) into a Pandas dataframe
- parse the mca into a dataframe
- clean the datasets (dedupe, add coords)

**Comment by james-westwood** [comment]
The clean `mca_schedule_df` looks good - lots of records with all `0` across the week though. 

image

**Comment by james-westwood** [comment]
The `mca_stop_df` also looks good

image

File: SDG_train_timetable.py [hunk 1]
```diff
@@ -0,0 +1,264 @@
...
+# Download link for zip
+# Name of zip changes each day so will need to scrape the page to extract
+# the latest
+# As such, providing hard-coding filepaths for the time being, and
+# a shortened mca file just for proof of concept.
+msn_file = os.path.join(output_directory, 'ttisf467.msn')
+mca_file = os.path.join(output_directory, 'ttisf467.mca')
+
+
+# Below code can be used once we have the latest filename
```
Comment (james-westwood) [inline SDG_train_timetable.py:31]: I think we should probably delete this. It's probably never going to be used.

File: SDG_train_timetable.py [hunk 2]
```diff
@@ -0,0 +1,264 @@
...
+train_timetable_zip_link = config["ENG_train_timetable_data"]
+output_directory = os.path.join(CWD, 'data', 'england_train_timetable')
+
+# Download link for zip
+# Name of zip changes each day so will need to scrape the page to extract
+# the latest
+# As such, providing hard-coding filepaths for the time being, and
+# a shortened mca file just for proof of concept.
+msn_file = os.path.join(output_directory, 'ttisf467.msn')
+mca_file = os.path.join(output_directory, 'ttisf467.mca')
```
Comment (james-westwood) [inline SDG_train_timetable.py:28]: Will these filenames ever change? If so, we should not hard code them. Better in the config.

**Review by james-westwood — APPROVED** [review]
I have followed the logic throughout - great comments by the way!

As you can see both dfs have the data in as expected

I've run the code through a linter so I don't think any further changes are needed.

**Comment by james-westwood** [comment]
@nkshaw23 This is ready to merge!

File: SDG_train_timetable.py (hunk 2 above)
Comment (nkshaw23) [reply SDG_train_timetable.py:28]: Added into config with instructions for the user added into the SDG method notes for future reference.

**Pull request merged**